service/cloudconfigv2/master_template.go

```diff
@@ -1,108 +1,108 @@
 package cloudconfigv2
 
 import (
-	"github.com/giantswarm/certificatetpr"
+	"github.com/giantswarm/certs"
 	k8scloudconfig "github.com/giantswarm/k8scloudconfig/v_1_1_0"
 	"github.com/giantswarm/microerror"
 )
 
 type masterExtension struct {
-	certs certificatetpr.AssetsBundle
+	certs certs.Cluster
 }
 
 func (e *masterExtension) Files() ([]k8scloudconfig.FileAsset, error) {
 	filesMeta := []k8scloudconfig.FileMetadata{
 		// Kubernetes API server.
 		{
-			AssetContent: string(e.certs[certificatetpr.AssetsBundleKey{certificatetpr.APIComponent, certificatetpr.CA}]),
+			AssetContent: string(e.certs.APIServer.CA),
 			Path:         "/etc/kubernetes/ssl/apiserver-ca.pem",
 			Owner:        FileOwner,
 			Permissions:  FilePermission,
 		},
 		{
-			AssetContent: string(e.certs[certificatetpr.AssetsBundleKey{certificatetpr.APIComponent, certificatetpr.Crt}]),
+			AssetContent: string(e.certs.APIServer.Crt),
 			Path:         "/etc/kubernetes/ssl/apiserver-crt.pem",
 			Owner:        FileOwner,
 			Permissions:  FilePermission,
 		},
 		{
-			AssetContent: string(e.certs[certificatetpr.AssetsBundleKey{certificatetpr.APIComponent, certificatetpr.Key}]),
+			AssetContent: string(e.certs.APIServer.Key),
 			Path:         "/etc/kubernetes/ssl/apiserver-key.pem",
 			Owner:        FileOwner,
 			Permissions:  FilePermission,
 		},
 		// Calico client.
 		{
-			AssetContent: string(e.certs[certificatetpr.AssetsBundleKey{certificatetpr.CalicoComponent, certificatetpr.CA}]),
+			AssetContent: string(e.certs.CalicoClient.CA),
 			Path:         "/etc/kubernetes/ssl/calico/client-ca.pem",
 			Owner:        FileOwner,
 			Permissions:  FilePermission,
 		},
 		{
-			AssetContent: string(e.certs[certificatetpr.AssetsBundleKey{certificatetpr.CalicoComponent, certificatetpr.Crt}]),
+			AssetContent: string(e.certs.CalicoClient.Crt),
 			Path:         "/etc/kubernetes/ssl/calico/client-crt.pem",
 			Owner:        FileOwner,
 			Permissions:  FilePermission,
 		},
 		{
-			AssetContent: string(e.certs[certificatetpr.AssetsBundleKey{certificatetpr.CalicoComponent, certificatetpr.Key}]),
+			AssetContent: string(e.certs.CalicoClient.Key),
 			Path:         "/etc/kubernetes/ssl/calico/client-key.pem",
 			Owner:        FileOwner,
 			Permissions:  FilePermission,
 		},
 		// Etcd client.
 		{
-			AssetContent: string(e.certs[certificatetpr.AssetsBundleKey{certificatetpr.EtcdComponent, certificatetpr.CA}]),
+			AssetContent: string(e.certs.EtcdServer.CA),
```

Comment:
So we had and still have only one key pair for etcd. There is an issue somewhere to use a dedicated key pair for the client as well. It just stays as it was.

Reply:
Yes, I also figured this out while simplify things. Simplifying things pays off :)